src/askui/models/askui/computer_agent.py

```diff
@@ -47,20 +71,26 @@ def _create_message(
         model_choice: str,  # noqa: ARG002
     ) -> MessageParam:
         try:
-            request_body = {
-                "max_tokens": self._settings.max_tokens,
-                "messages": [msg.model_dump(mode="json") for msg in messages],
-                "model": self._settings.model,
-                "tools": self._tool_collection.to_params(),
-                "betas": self._settings.betas,
-                "system": [self._system],
-            }
+            request_body = RequestBody(
+                max_tokens=self._settings.max_tokens,
+                messages=messages,
+                model=self._settings.model,
+                tools=self._tool_collection.to_params(),
+                betas=self._settings.betas,
+                system=[self._system],
+                tool_choice=self._settings.tool_choice,
+            )
+            if self._settings.thinking:
+                request_body.thinking = ThinkingConfigEnabledParam(
+                    budget_tokens=self._settings.thinking.budget_tokens,
+                    type="enabled",
+                )
+
             response = self._client.post(
-                "/act/inference", json=request_body, timeout=300.0
+                "/act/inference", json=request_body.model_dump(), timeout=300.0
             )
             response.raise_for_status()
-            response_data = response.json()
-            return MessageParam.model_validate(response_data)
+            return MessageParam.model_validate(response.json())
```

Comment:
```suggestion
            return MessageParam.model_validate_json(response.text)
```

https://docs.pydantic.dev/latest/concepts/performance/#in-general-use-model_validate_json-not-model_validatejsonloads